### DISCUSSION TITLE
Changed behavior of BeEquivalentTo after update to version 6.12.2

### DISCUSSION THREAD

**Discussion by kroknstiTE** [fluentassertions/fluentassertions#2810] [Q&A] [answered]
After updating to version 6.12.2. some of my tests failed.

In my comparison I have a entity with explicit implementation of a interface (ITenant with Guid TenantId) and a backing field.
Something like this:
```
private readonly Guid tenantId = Guid.Empty;
Guid ITenant.TenantId => tenantId;
```
This is populated by Ef Core infrastructure,  I do not touch this in application logic.

After update nuget,  comparing a entity implementing this interface to another instance gives comparison error due to mismatch on TenantId. The manually created instance for comparison has TenantId = Guid.Empty. Of course one could argue that I should initialize it etc... but how could i exclude this member in my comparison? I tried some strategies without success:

```
options.Excluding(t => ((ITenant)t).TenantId) 
options.Excluding(t => t.Name == "TenantId")
```
Note that I do not think the observed behavior is wrong. Just asking how I could exclude this member.

**Comment by dennisdoomen** [comment]
Yes, 6.12.2 was specifically meant to fix the explicit implementation property support, so I'm sorry it now surfaces another issue for you.

The best would be to use the second workaround, but I think the name will be something along the lines of `ITenant.TenantId`.

**Reply by stiankroknes** [reply]
No problem. I used the second workaround, but I also had to include the namespace as part of the name. Thanks!

**Comment by stiankroknes** [comment, marked as answer]
This exclude did the trick:
```
.Excluding(p => p.Name == $"{typeof(ITenant).Namespace}.{nameof(ITenant)}.{nameof(ITenant.TenantId)}");
```
